src/Sulu/Bundle/MediaBundle/Media/Storage/FlysystemStorage.php

```diff
@@ -60,41 +63,47 @@ public function save(string $tempPath, string $fileName, array $storageOptions =
             throw new FilenameAlreadyExistsException($filePath);
         }
 
-        return [
-            'segment' => $segment,
-            'fileName' => $fileName,
-        ];
+        return $storageOptions;
     }
 
     public function load(array $storageOptions)
     {
-        $segment = $this->getStorageOption($storageOptions, 'segment');
-        $fileName = $this->getStorageOption($storageOptions, 'fileName');
-        $path = $segment . '/' . $fileName;
+        $filePath = $this->getFilePath($storageOptions);
 
         try {
-            return $this->filesystem->readStream($path);
+            return $this->filesystem->readStream($filePath);
         } catch (FileNotFoundException $exception) {
-            throw new ImageProxyMediaNotFoundException(\sprintf('Original media at path "%s" not found', $path));
+            throw new ImageProxyMediaNotFoundException(\sprintf('Original media at path "%s" not found', $filePath));
         }
     }
 
     public function remove(array $storageOptions): void
     {
-        $segment = $this->getStorageOption($storageOptions, 'segment');
-        $fileName = $this->getStorageOption($storageOptions, 'fileName');
+        $filePath = $this->getFilePath($storageOptions);
 
-        if (!$segment || !$fileName) {
+        if (!$filePath) {
             return;
         }
 
         try {
-            $this->filesystem->delete($segment . '/' . $fileName);
+            $this->filesystem->delete($filePath);
         } catch (FileNotFoundException $exception) {
         }
     }
 
-    protected function getUniqueFileName(string $folder, string $fileName, int $counter = 0): string
+    public function move(array $sourceStorageOptions, array $targetStorageOptions): void
+    {
+        $this->createDirectories($targetStorageOptions);
+
+        $targetFilePath = $this->getFilePath($targetStorageOptions);
+        if ($this->filesystem->has($targetFilePath)) {
```

Comment:
as discussed we should resolve the conflict and return here the new storageOptions